cond/cond.go

```diff
@@ -88,15 +88,27 @@ func (c *Cond) ContextWait(ctx context.Context) bool {
 }
 
 func (c *Cond) cancelWait() {
+	gotLock := make(chan struct{})
+	cancelled := false
+
+	go func() {
+		c.L.Lock()
+		if cancelled {
```

Comment:
There’s a data race on `cancelled`.

It’s possible for the Signal to receive the cancel and for the client to unlock, both before it was able to set the `cancelled` boolean.

I added a test to your branch that shows this.

Reply:
Nice catch.  Thanks, fixed.

Reply:
Ah! Good job! I was trying to make it work, but was failing miserably.
I fixed my test, it's not simpler and more meaningful.
I forced-pushed to your branch.